tests/Doctrine/Tests/DBAL/Functional/WriteTest.php

```diff
@@ -135,20 +138,46 @@ public function testUpdate() : void
 
     public function testLastInsertId() : void
     {
-        if (! $this->connection->getDatabasePlatform()->prefersIdentityColumns()) {
-            $this->markTestSkipped('Test only works on platforms with identity columns.');
+        if (! $this->connection->getDatabasePlatform()->supportsIdentityColumns()) {
+            $this->markTestSkipped('This test targets platforms that support identity columns.');
         }
 
         self::assertEquals(1, $this->connection->insert('write_table', ['test_int' => 2, 'test_string' => 'bar']));
-        $num = $this->lastInsertId();
+        $num = $this->connection->lastInsertId();
 
         self::assertGreaterThan(0, $num, 'LastInsertId() should be non-negative number.');
     }
 
-    public function testLastInsertIdSequence() : void
+    public function testLastInsertIdNotSupported() : void
     {
-        if (! $this->connection->getDatabasePlatform()->supportsSequences()) {
-            $this->markTestSkipped('Test only works on platforms with sequences.');
+        if ($this->connection->getDatabasePlatform()->supportsIdentityColumns()) {
+            $this->markTestSkipped('This test targets platforms that don\'t support identity columns.');
+        }
+
+        self::assertEquals(1, $this->connection->insert('write_table', ['test_int' => 2, 'test_string' => 'bar']));
```

Comment:
Isn't this redundant?

Reply:
I would keep it, as `lastInsertId()` may fail for 2 reasons:

- identity columns not supported
- no previous statement has inserted a row

By inserting a row first in the same test, we can guarantee that we're actually testing the first bullet point.

Reply:
That's why I'd rather have different exception classes and expect a specific class. If this is out of scope, I'd rather expect a specific error message on the exception than perform an unrelated operation that still doesn't guarantee that the exception is thrown by the code that is intended to be tested.

Reply:
Makes sense. This takes us back to my suggested modification to driver exceptions, but with different subclassing (I originally intended to have a single `DriverException` class).

Exceptions would not be based on the driver (`DB2Exception`) anymore, but on the intent (`IdentityColumnNotSupportedException`).

This would definitely be clearer. But apart from tests, does this have a real-life use case?